text/char-rnn/char-rnn.jl

```diff
@@ -49,22 +51,25 @@ end
 
 function getdata(args)
     ## Download the data if not downloaded as 'input.txt'
-    isfile("input.txt") ||
-        download("https://cs.stanford.edu/people/karpathy/char-rnn/shakespeare_input.txt","input.txt")
+    isfile("input.txt") || download(
+        "https://cs.stanford.edu/people/karpathy/char-rnn/shakespeare_input.txt",
+        "input.txt",
+    )
+
+    text = String(read("input.txt"))
 
-    text = collect(String(read("input.txt")))
-    
     ## an array of all unique characters
     alphabet = [unique(text)..., '_']
-    
-    text = map(ch -> onehot(ch, alphabet), text)
-    stop = onehot('_', alphabet)
+    stop = '_'
 
     N = length(alphabet)
     
-    ## Partitioning the data as sequence of batches, which are then collected as array of batches
-    Xs = collect(partition(batchseq(chunk(text, args.nbatch), stop), args.seqlen))
-    Ys = collect(partition(batchseq(chunk(text[2:end], args.nbatch), stop), args.seqlen))
+    ## Partitioning the data as sequence of batches, which are then collected 
+    ## as array of batches
+    Xs = partition(batchseq(chunk(text, args.batchsz), stop), args.seqlen)
+    Ys = partition(batchseq(chunk(text[2:end], args.batchsz), stop), args.seqlen)
+    Xs = [Flux.onehotbatch.(bs, (alphabet,)) for bs in Xs]
+    Ys = [Flux.onehotbatch.(bs, (alphabet,)) for bs in Ys]
```

Comment:
Is it possible to rely on `onehotbatch`es handling of higher-dim arrays for this instead of broadcasting? Otherwise LGTM.

Reply:
I messed with this a little, and it is kind of rough to go from the code here to higher dimensional one hot batches. I can come back to it later, but unless someone else has a quick fix for it now, I'd like to submit if everything else is ok.